### ISSUE TITLE
Modify webapp backend to process all user input as files

### ISSUE THREAD

**Issue by talumbau** [OpenSourcePolicyCenter/PolicyBrain#619]
With the merge of https://github.com/open-source-economics/Tax-Calculator/pull/1505, Tax-Calculator now supports a JSON reform interface that maps very naturally to the TaxBrain GUI interface. Currently, TaxBrain supports two separate but similar workflows: a GUI interface and a file upload interface. For the GUI interface, there is a significant amount of code that processes the user's input to create a job for submission to the worker nodes. There is a separate path of control to take a file-based input and submit that to the worker nodes. A way to reduce complexity of TaxBrain would be to unify these flows. To do this, code would be written to transform a user's input to a JSON reform (represented in memory as a dictionary) in the style of https://github.com/open-source-economics/Tax-Calculator/pull/1505. For example, this user input:

<img width="762" alt="screen shot 2017-08-07 at 9 41 03 pm" src="https://user-images.githubusercontent.com/1891418/29242492-408ebc0a-7f43-11e7-9a8c-b6118be8d66f.png">

Would be something like:

```
{
 "policy": {
   "_SS_thd50_joint": {"2019": [36000]}
 }
}
```

Once this works is completed, the workflows would look like this:

File based reform:
 - take file from user POST and store in memory (already working)
 - submit file contents to worker nodes (already working)

GUI-based reform:
 - take params from user POST (already working)
 - create in-memory representation of reform file based on user input (NEW WORK HERE)
 - submit file contents to work nodes (already working)

After the new work is done, a lot of code could be removed from TaxBrain (e.g. the `expand_arrays` functions, `package_up_vars`, etc.) and there would only be one flow of control for submitting jobs to the worker nodes.

cc @MattHJensen, @brittainhard, @PeterDSteinberg, @martinholmer

**Comment by martinholmer** [comment]
Let me add something to the #619 comment by @talumbau.
In addition to the TaxBrain simplifications outlined by @talumbau, the merge of [Tax-Calculator pull request 1502](https://github.com/open-source-economics/Tax-Calculator/pull/1502) and [Tax-Calculator pull request 1503](https://github.com/open-source-economics/Tax-Calculator/pull/1503) offers another TaxBrain simplification: **there is no longer a need for TaxBrain to do any policy parameter validations**.

cc @MattHJensen, @hdoupe, @brittainhard, @PeterDSteinberg

**Comment by martinholmer** [comment]
Let me add something else to the #619 comment by @talumbau.
While #619 is labeled an enhancement, when implemented the ideas of @talumbau will fix at least one bug (#596) and perhaps other bugs as well.

cc @MattHJensen, @hdoupe, @brittainhard, @PeterDSteinberg

**Comment by martinholmer** [comment]
@brittainhard and @PeterDSteinberg,
It has been several days since @talumbau suggested the TaxBrain simplification described in #619, but neither of you have commented on the pros and cons of his suggestion.  We would like to hear your views on his suggestion.  Can you add your comments to #619?

@MattHJensen @GoFroggyRun @hdoupe

**Comment by hdoupe** [comment]
@martinholmer said 
> Let me add something else to the #619 comment by @talumbau. While #619 is labeled an enhancement, when implemented the ideas of @talumbau will fix at least one bug (#596) and perhaps other bugs as well. 

I agree.  I think this would take care of the TaxBrain GUI input bugs (#596, #598, #607, and #609).

**Comment by martinholmer** [comment]
TaxBrain 1.0.3, which is using Tax-Calculator 0.10.2, uses the new approach to processing the GUI input that @talumbau suggested in #619.  @talumbau, thanks for the strategy, and @hdoupe, thanks for the 
implementation.  Closing #619.

@MattHJensen @GoFroggyRun
